internal/namespaces/instance/v1/custom_server_create_builder.go

```diff
@@ -314,3 +355,220 @@ func (sb *ServerBuilder) Validate() error {
 func (sb *ServerBuilder) Build() (*instance.CreateServerRequest, *instance.CreateIPRequest) {
 	return sb.createReq, sb.createIPReq
 }
+
+type PostServerCreationSetupFunc func(ctx context.Context, server *instance.Server) error
+
+func (sb *ServerBuilder) BuildPostCreationSetup() PostServerCreationSetupFunc {
+	rootVolume := sb.rootVolume
+	volumes := sb.volumes
+
+	return func(ctx context.Context, server *instance.Server) error {
+		if rootVolume != nil {
+			serverRootVolume, hasRootVolume := server.Volumes["0"]
+			if !hasRootVolume {
+				return errors.New("root volume not found")
+			}
+			rootVolume.ExecutePostCreationSetup(ctx, sb.apiBlock, serverRootVolume.ID)
+		}
+		for i, volume := range volumes {
+			serverVolume, serverHasVolume := server.Volumes[strconv.Itoa(i+1)]
+			if !serverHasVolume {
+				return fmt.Errorf("volume %d not found in server", i+1)
+			}
+			volume.ExecutePostCreationSetup(ctx, sb.apiBlock, serverVolume.ID)
+		}
+
+		return nil
+	}
+}
+
+type VolumeBuilder struct {
+	Zone       scw.Zone
+	VolumeType instance.VolumeVolumeType
+
+	// SnapshotID is the ID of the snapshot the volume should be created from.
+	SnapshotID *string
+	// VolumeID is the ID of the volume if one should be imported.
+	VolumeID *string
+	// Size is the size of the created Volume. If used, the volume should be created from scratch.
+	Size *scw.Size
+	// IOPS is the io per second to be configured for a created volume.
+	IOPS *uint32
+}
+
+// NewVolumeBuilder creates a volume builder from a 'volumes' argument item.
+//
+// Volumes definition must be through multiple arguments (eg: volumes.0="l:20GB" volumes.1="b:100GB" volumes.2="sbs:50GB:15000)
+//
+// A valid volume format is either
+// - a "creation" format: ^((local|l|block|b|scratch|s|sbs):)?\d+GB?(:\d+)?$ (size is handled by go-humanize, so other sizes are supported)
+// - a "creation" format with a snapshot id: l:<uuid> b:<uuid>
+// - a UUID format
+func NewVolumeBuilder(zone scw.Zone, flagV string) (*VolumeBuilder, error) {
+	parts := strings.Split(strings.TrimSpace(flagV), ":")
+	vb := &VolumeBuilder{
+		Zone: zone,
+	}
+
+	if len(parts) == 3 {
+		iops, err := strconv.ParseUint(parts[2], 10, 32)
+		if err != nil {
+			return nil, fmt.Errorf("invalid volume iops %s in %s volume", parts[2], flagV)
+		}
+		vb.IOPS = scw.Uint32Ptr(uint32(iops))
+		parts = parts[0:2]
+	}
+
+	if len(parts) == 2 {
+		switch parts[0] {
+		case "l", "local":
```

Comment:
Could you add documentation for all types of volumes, it seems complicated to find the right meaning without examples and knowing what are the different options available for a given server.